daemon/src/model/cfd.rs

```diff
@@ -2456,18 +2533,165 @@ mod tests {
             )
         }
 
-        fn dummy_open(event_id: BitMexPriceEventId) -> Self {
-            let cfd = Cfd::from_order(
+        fn maker_short() -> Self {
+            Cfd::from_order(
+                Order::dummy_model(),
+                Position::Short,
+                Usd::new(dec!(1000)),
+                dummy_identity(),
+                Role::Maker,
+            )
+        }
+
+        fn dummy_not_open_yet() -> Self {
+            Cfd::from_order(
                 Order::dummy_model(),
                 Position::Long,
                 Usd::new(dec!(1000)),
                 dummy_identity(),
                 Role::Taker,
-            );
+            )
+        }
 
+        fn dummy_open(self, event_id: BitMexPriceEventId) -> Self {
             Event::dummy_open(event_id)
                 .into_iter()
-                .fold(cfd, Cfd::apply)
+                .fold(self, Cfd::apply)
+        }
+
+        fn with_lock(mut self) -> Self {
+            let (sk_taker, pk_taker) = crate::keypair::new(&mut rand::thread_rng());
+            let (sk_maker, pk_maker) = crate::keypair::new(&mut rand::thread_rng());
+
+            let (taker_margin, maker_margin, own_sk) = match self.role {
+                Role::Maker => (self.counterparty_margin(), self.margin(), sk_maker),
+                Role::Taker => (self.margin(), self.counterparty_margin(), sk_taker),
+            };
+
+            let taker_params = BuildPartyParams {
+                amount: taker_margin,
+                identity_pk: pk_taker,
+                fee_rate: Default::default(),
+            };
+
+            let maker_params = BuildPartyParams {
+                amount: maker_margin,
+                identity_pk: pk_maker,
+                fee_rate: Default::default(),
+            };
+
+            let taker_params = build_party_params(taker_params).unwrap();
+            let maker_params = build_party_params(maker_params).unwrap();
+
+            let taker_address = taker_params.address.clone();
+            let maker_address = maker_params.address.clone();
+
+            let (lock_tx, descriptor) = build_lock_tx(maker_params, taker_params);
```

Comment:
Could the problem be that the lock transaction stored in the Dlc is not signed? I thought that should not matter as long as we don't publish on chain 🤔